File: 406. Queue Reconstruction by Height.cpp
/*
	1. 根据身高从大到小排序，如果身高相同，根据k值从小到大排序
	2. 根据k值插入到结果列表的k位置
*/
class Solution {
public:
    static bool cmp(const pair<int, int>& p1, const pair<int, int>& p2)
    {
        if (p1.first != p2.first) return p1.first > p2.first;
        return p1.second < p2.second;
    }
    
    vector<pair<int, int>> reconstructQueue(vector<pair<int, int>>& people) {
        sort(people.begin(), people.end(), cmp);
        vector<pair<int, int>> res;
        for (int i = 0; i < people.size(); i++)
        {
            // printf("(%d, %d), ", people[i].first, people[i].second);
            int pos = people[i].second;
            res.insert(res.begin() + pos, people[i]);
        }
        return res;
    }
};

File: python/406. Queue Reconstruction by Height.py
class Solution(object):
    def reconstructQueue(self, people):
        """
        :type people: List[List[int]]
        :rtype: List[List[int]]
        """
        people.sort(key=lambda x: (-x[0], x[1]))
        ans = []
        for h, k in people:
            ans.insert(k, (h, k))
        return ans

# 思路：按照身高降序，K升序排序
#       先将身高高的排好，
